<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="tower_pitch">

<!-- PROPERTIES -->


<!-- GEOMETRY -->
    <xacro:macro name="tower_pitch_geometry">
        <geometry>
            <box size="${tower_pitch_length} ${tower_pitch_width} ${tower_pitch_height}"/>
        </geometry>
    </xacro:macro>

<!-- MATERIAL -->
    <xacro:macro name="tower_pitch_material">
        <material name="light_gray">
            <color rgba=".3 .3 .5 1"/>
        </material>
    </xacro:macro>

<!-- LINK -->
    <xacro:macro name="tower_pitch" params="name parent x y z">
        <link name="${name}">
            <visual>
                <xacro:tower_pitch_geometry/>
                <xacro:tower_pitch_material/>
            </visual>
            <collision>
                <xacro:tower_pitch_geometry/>
            </collision>
            <inertial>
                <mass value="1"/>
                <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
            </inertial>
        </link>

            <joint name="${name}_joint" type="revolute">
                <parent link="${parent}"/>
                <child link="${name}"/>
                <origin rpy="0 0 0" xyz="${x} ${y} ${z}"/>
                <axis xyz="0 1 0"/>
                <limit effort="100" velocity="10" lower="-0.785" upper="0.785"/>
                <dynamics damping="1"/>
            </joint>

            <gazebo reference="${name}">
                <mu1> .2 </mu1>
                <mu2> .2 </mu2>
                <material> Gazebo/Wh </material>
                <selfCollide> True </selfCollide>
            </gazebo>

            <gazebo>
                <plugin name="joint_limiter_${name}" filename="libJointLimiter.so">
                    <joint>${name}_joint</joint>
                    <topic>/simulation/controller/vel/${name}</topic>
                    <lower>-0.785</lower>
                    <upper>0.785</upper>
                </plugin>
            </gazebo>
            <gazebo>
                <plugin name="joint_controller_${name}" filename="libControllerPID.so">
                    <joint>${name}_joint</joint>
                    <topic>/simulation/controller/vel/${name}</topic>
                    <Kp>0.8</Kp>
                    <Ki>0.25</Ki>
                    <Kd>0.1</Kd>
                </plugin>
            </gazebo>


            <!--transmission name="${name}_transmission">
                <type>transmission_interface/SimpleTransmission</type>
                <joint name="${name}_joint">
                    <hardwareInterface>EffortJointInterface</hardwareInterface>
                </joint>
                <actuator name="${name}_motor">
                    <hardwareInterface>EffortJointInterface</hardwareInterface>
                    <mechanicalReduction>1</mechanicalReduction>
                </actuator>
            </transmission-->

    </xacro:macro>

</robot>
